docs/containers/docker-compose.md

```diff
@@ -193,6 +193,76 @@ volumes:
     - /host-folder-path:/container-folder-path
 ```
 
+## Docker Compose with multiple Compose files
+Workspaces can have multiple docker-compose files to handle different environments like development, test and production. The content of the configuration can be split into multiple files. For example, a base compose file that defines the common information for all environments and override files that defines environment specific information. When these files are passes as input to `docker-compose` command, it will combine these files into single configuration. By default the **Docker: Compose Up** command will pass a single file as input to compose command. But the compose command can be customized to pass in multiple files using [command customization](https://code.visualstudio.com/docs/containers/reference#_command-customization). This can be also be achieved using [custom task](https://code.visualstudio.com/docs/editor/tasks#_custom-tasks) to invokes the `docker-compose` command with desired parameters.
+
+Note: If your workspace has `docker-compose.yml` and `docker-compose.override.yml` and no other compose files, then the `docker-compose` command will be invoked with no input files and it will implicitly use these files. In this case, no customization is needed.
+### Command Customization
+[Command customization](https://code.visualstudio.com/docs/containers/reference#_command-customization) provides various ways to customize the compose up based on your requirement. The following are few sample command customization for compose up command.
+#### Base file and an override file
+Let's assume your workspace has a base compose file (`docker-compose.yml`) and an override file for each environments (`docker-compose.dev.yml`, `docker-compose.test.yml` and `docker-compose.prod.yml`) and you always compose up with base file and an override file. In this case the compose up command can be customized like below. When the compose up command is invoked the ${configuration} will be replaced by the selected file.
+```json
+"docker.commands.composeUp": [
+    {
+        "label": "override",
+        "template": "docker-compose -f docker-compose.yml ${configurationFile}  up -d --build",
+    }
+]
+```
+#### Template matching
+Let's assume you have different set of input files for each environments. You could define multiple templates with regular expression match, and the selected file name will be matched against this `match` property and corresponding template will be used.
```

Comment:
"have different" should be "have a different"
"each environments" should be "each environment"
"and corresponding" should be "and the corresponding"